onadata/apps/api/viewsets/user_profile_viewset.py

```diff
@@ -100,7 +100,8 @@ def serializer_from_settings():
 
 
 def set_is_email_verified(profile, is_email_verified):
-    profile.metadata.update({"is_email_verified": is_email_verified})
+    profile.refresh_from_db()
```

Comment:
Can we make this call before we call `set_is_email_verified()`? I see a reason why the `refresh_from_db` may need to happen on the GET request to verify, but perhaps not on the POST request. I would hope all requests on a POST use the master DB. 

Is this issue due to a read replica update delay?

Reply:
Yes, I can change this.

Reply:
Doesn't seem to be a read replica update delay; the change never shows up. Noticed accounts that had been verified a while back(2-3 days) but the metadata wasn't updated

Reply:
What then guarantees this issue is resolve considering the statement has always been there? The only new statements is the cache deletion and the refresh_from_db.

Reply:
Changed this in the latest commits. The main issue seems to have been the `RegistrationProfile` retrieval in the user profile viewset.